pageserver/src/layered_repository.rs

```diff
@@ -1004,11 +1099,11 @@ impl Timeline for LayeredTimeline {
     }
 
     fn get_start_lsn(&self) -> Lsn {
-        self.ancestor_timeline
-            .as_ref()
-            .and_then(|ancestor_entry| ancestor_entry.local_or_schedule_download(self.tenantid))
-            .map(Timeline::get_start_lsn)
-            .unwrap_or(self.ancestor_lsn)
+        // TODO ask @knizhnik about this change,
+        // previously get_start_lsn always returned zero because it was recursive
+        // and timeline without an ancestor has 0 ancestor lsn
+        // Can it be removed in favor of self.get_ancestor_lsn?
+        self.ancestor_lsn
```

Comment:
@knizhnik Please see the comment in code. Am I correct?

Reply:
It seems to be correct.
But I wonder why you are asking me - it's not my code:)

Reply:
Sorry if I've misinterpreted something, but git blame points to this commit: [#b227c63ed](https://github.com/zenithdb/zenith/commit/b227c63edf1c7d3622daeb3b47ba693c84430052#) this is when this method was introduced

Reply:
O, I see. get_start_lsn() method was really introduced by me.
But then this code was refactored so that I even do not recognize it as my code.

But in any case, I think it is safe to use ancestor_lsn without recursion.